test/unit/models/classes/clientConfig/ClientConfigStorageTest.php

```diff
@@ -0,0 +1,120 @@
+{
+    /** @var TokenService|MockObject */
+    private TokenService $tokenService;
+
+    /** @var ClientLibRegistry|MockObject */
+    private ClientLibRegistry $clientLibRegistry;
+
+    /** @var FeatureFlagConfigSwitcher|MockObject */
+    private FeatureFlagConfigSwitcher $featureFlagConfigSwitcher;
+
+    /** @var AssetService|MockObject */
+    private AssetService $assetService;
+
+    /** @var common_ext_ExtensionsManager|MockObject */
+    private common_ext_ExtensionsManager $extensionsManager;
+
+    /** @var ClientConfigService|MockObject */
+    private ClientConfigService $clientConfigService;
+
+    /** @var UserLanguageServiceInterface|MockObject */
+    private UserLanguageServiceInterface $userLanguageService;
+
+    /** @var FeatureFlagRepositoryInterface|MockObject */
+    private FeatureFlagRepositoryInterface $featureFlagRepository;
+
+    /** @var ResolverFactory|MockObject */
+    private ResolverFactory $resolverFactory;
+
+    /** @var LoggerInterface|MockObject */
+    private LoggerInterface $logger;
+
+    protected function setUp(): void
+    {
+        $this->tokenService = $this->createMock(TokenService::class);
+        $this->clientLibRegistry = $this->createMock(ClientLibRegistry::class);
+        $this->featureFlagConfigSwitcher = $this->createMock(FeatureFlagConfigSwitcher::class);
+        $this->assetService = $this->createMock(AssetService::class);
+        $this->extensionsManager = $this->createMock(common_ext_ExtensionsManager::class);
+        $this->clientConfigService = $this->createMock(ClientConfigService::class);
+        $this->userLanguageService = $this->createMock(UserLanguageServiceInterface::class);
+        $this->featureFlagRepository = $this->createMock(FeatureFlagRepositoryInterface::class);
+        $this->resolverFactory = $this->createMock(ResolverFactory::class);
+        $this->logger = $this->createMock(LoggerInterface::class);
+
+        $this->sut = new ClientConfigStorage(
+            $this->tokenService,
+            $this->clientLibRegistry,
+            $this->featureFlagConfigSwitcher,
+            $this->assetService,
+            $this->extensionsManager,
+            $this->clientConfigService,
+            $this->userLanguageService,
+            $this->featureFlagRepository,
+            $this->resolverFactory,
+            $this->logger
+        );
+    }
+
+    public function testGetConfig(): void
+    {
+        $query = $this->createMock(GetConfigQuery::class);
+        $query
+            ->method('getExtension')
+            ->willReturn('extension');
+        $query
+            ->method('getAction')
+            ->willReturn('action');
+        $query
+            ->method('getModule')
+            ->willReturn('module');
+        $query
+            ->method('getShownExtension')
+            ->willReturn('shownExtension');
+        $query
+            ->method('getShownStructure')
+            ->willReturn('shownStructure');
+    }
+}
```

Comment:
- [ ] Remember to extend this test or call `markTestIncomplete()`

```suggestion
        $query
            ->method('getShownStructure')
            ->willReturn('shownStructure');
            
        $this->markTestIncomplete();
    }
}
```

Reply:
I think it is in progress yet by @shpran . We should test at least the happy path here.